tests/components/test_rest_command.py

```diff
@@ -222,21 +222,82 @@ def test_rest_command_put(self, aioclient_mock):
         assert len(aioclient_mock.mock_calls) == 1
         assert aioclient_mock.mock_calls[0][2] == b'data'
 
-    def test_rest_command_content_type(self, aioclient_mock):
-        """Call a rest command with a content type."""
-        data = {
-            'payload': 'item',
-            'content_type': 'text/plain'
+    def test_rest_command_headers(self, aioclient_mock):
+        """Call a rest command with custom headers and content types."""
+        header_config_variations = {
+            rc.DOMAIN: {
+                'no_headers_test': {},
+                'content_type_test': {
+                    'content_type': 'text/plain'
+                },
+                'headers_test': {
+                    'headers': {
+                        'Accept': 'application/json',
+                        'User-Agent': 'Mozilla/5.0'
+                    }
+                },
+                'headers_and_content_type_test': {
+                    'headers': {
+                        'Accept': 'application/json'
+                    },
+                    'content_type': 'text/plain'
+                },
+                'headers_and_content_type_override_test': {
+                    'headers': {
+                        'Accept': 'application/json',
+                        aiohttp.hdrs.CONTENT_TYPE: 'application/pdf'
+                    },
+                    'content_type': 'text/plain'
+                }
+            }
         }
-        self.config[rc.DOMAIN]['post_test'].update(data)
 
-        with assert_setup_component(4):
-            setup_component(self.hass, rc.DOMAIN, self.config)
+        # add common parameters
+        for variation in header_config_variations[rc.DOMAIN].values():
+            variation.update({'url': self.url, 'method': 'post',
+                              'payload': 'test data'})
 
+        with assert_setup_component(5):
+            setup_component(self.hass, rc.DOMAIN, header_config_variations)
+
+        # provide post request data
         aioclient_mock.post(self.url, content=b'success')
 
-        self.hass.services.call(rc.DOMAIN, 'post_test', {})
-        self.hass.block_till_done()
+        for test_service in ['no_headers_test',
```

Comment:
```for test_service in header_config_variations[rc.DOMAIN].keys():```

Reply:
I considered something like that, but the ordering of keys() won't be well defined without using sorted() will it? The order however is relied upon for the assertions to be able to index into the mocked responses. 

Alphabetical order from sorted() is not the most helpful order to apply the assertions so then you'd end up with the assertions not using consecutive indices. I preferred to explicitly define the order from simple->complex in order to be more specific if it caught an error and felt that was cleaner code as well.